lib/puppet_x/puppetlabs/cd4pe_client.rb

```diff
@@ -186,7 +196,7 @@ def web_ui_endpoint
       endpoint
     end
 
-    def make_request(type, api_url, payload = '')
+    def make_request(type, api_url, payload = '', target_file = nil)
```

Comment:
The `target_file` parameter probably doesn't belong here, with code that's about making web requests.

I would leave the response handling to the caller, so this method retains the tight cohesion.

Especially since it's just as easy to call `create_and_write_tempfile` at the call site rather than in here.

Reply:
bruh. Genius